src/test/groovy/org/prebid/server/functional/tests/BidExpResponseSpec.groovy

```diff
@@ -356,9 +299,350 @@ class BidExpResponseSpec extends BaseSpec {
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        def response = pbsEmptyTtlService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain exp data"
+        assert response.seatbid.first.bid.first.exp == accountCacheTtl
+
+        where:
+        mediaType | accountCacheTtl       | accountAuctionConfig
+        BANNER    | PBSUtils.randomNumber | new AccountAuctionConfig(bannerCacheTtl: accountCacheTtl)
+        VIDEO     | PBSUtils.randomNumber | new AccountAuctionConfig(videoCacheTtl: accountCacheTtl)
+    }
+
+    def "PBS auction should resolve bid.exp when cache.{banner/video}-ttl-seconds config specified"() {
+        given: "Default bid request"
+        def bidRequest = BidRequest.getDefaultBidRequest().tap {
+            imp[0] = Imp.getDefaultImpression(mediaType)
+            enableCache()
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = pbsOnlyHostCacheTtlService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain exp data"
+        assert response.seatbid.first.bid.first.exp == expValue
+
+        where:
+        mediaType | expValue
+        BANNER    | BANNER_TTL_HOST_CACHE
+        VIDEO     | VIDEO_TTL_HOST_CACHE
+    }
+
+    def "PBS auction shouldn't resolve bid.exp when cache ttl-seconds is specified for #mediaType mediaType request"() {
+        given: "Default bid request"
+        def bidRequest = BidRequest.getDefaultBidRequest().tap {
+            imp[0] = Imp.getDefaultImpression(mediaType)
+            ext.prebid.cache = new PrebidCache(bids: new PrebidCacheSettings(ttlSeconds: PBSUtils.randomNumber))
+        }
+
+        when: "PBS processes auction request"
+        def response = pbsOnlyHostCacheTtlService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response shouldn't contain exp data"
+        assert !response.seatbid.first.bid.first.exp
+
+        where:
+        mediaType << [NATIVE, AUDIO]
+    }
+
+    def "PBS auction should resolve bid.exp when cache.default-ttl-seconds.{banner,video,audio,native} is specified and no higher-priority fields are present"() {
+        given: "Default bid request"
+        def bidRequest = BidRequest.getDefaultBidRequest().tap {
+            imp[0] = Imp.getDefaultImpression(mediaType)
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = pbsServiceFactory
+                .getService(EMPTY_CACHE_TTL_HOST_CONFIG + DEFAULT_CACHE_TTL_CONFIG)
+                .sendAuctionRequest(bidRequest)
```

Comment:
Not like this, because you still have an open container in memory, and it’s possible that you will face issues with unexpected closing of the active one.
Please add a configuration step and a cleanup:
```
cleanup:
    pbsServiceFactory.removeContainer(config)
```
as example:
```
PBS should include analytics tag in response when request and account allow client details but default doesn't
```